src/core/Form/Checkbox/Checkbox.baseStyles.tsx

```diff
@@ -0,0 +1,175 @@
+import { css } from 'styled-components';
+import {
+  withSuomifiTheme,
+  TokensAndTheme,
+  SuomifiThemeProp,
+} from '../../theme';
+import { disabledCursor } from '../../../components/utils/css';
+import { element, font } from '../../theme/reset';
+
+/* stylelint-disable */
+const checkedStyles = ({ theme }: SuomifiThemeProp) => css`
+  &.fi-checkbox--checked .fi-checkbox_label {
+    &::before,
+    &::after {
```

Comment:
Are these the styles that prevent using current focus styles with the component? Are these referring to the container? If so, shouldn't it be possible to target these styles without pseudo and use the current focus styles?

Reply:
These are referring to the pseudo elements of the label. Before is the checkbox itself and after was the checkmark. After is now gone after the change to the checkmark icon, but  my suspicion is that applying the focus styles on pseudo elements breaks them. Design says the focus should be around the checkbox only. works fine if applied to the whole component. Need to investigate more now that one of the pseudo elements is gone.

Reply:
Do I understand correctly by assuming that it makes no difference in this case if we don't use the pseudo focus styles since z-index should have no effect on visuals?

Reply:
In this case we are not able to use the pseudo focus styles. We can use the non-pseudo box-shadow variety just fine after it's implemented.